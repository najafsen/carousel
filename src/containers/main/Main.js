import React from "react";
import "./Main.scss";
import { CarouselSlide } from "../../components/carousel/CarouselSlide";
import { Carousel } from "../../components/carousel/Carousel";
import { Slide1 } from "./slides/slide1/Slide1";
import { Slide2 } from "./slides/slide2/Slide2";
import { Slide3 } from "./slides/slide3/Slide3";
import { Slide4 } from "./slides/slide4/Slide4";
import { Slide5 } from "./slides/slide5/Slide5";

export const Main = () => {
  return (
    <div className="main-component">
      <div className="header-container"></div>
      <div className="carousel-container">
        <Carousel>
          <CarouselSlide>
            <Slide1 />
          </CarouselSlide>
          <CarouselSlide>
            <Slide2 />
          </CarouselSlide>
          <CarouselSlide>
            <Slide3 />
          </CarouselSlide>
          <CarouselSlide>
            <Slide4 />
          </CarouselSlide>
          <CarouselSlide>
            <Slide5 />
          </CarouselSlide>
        </Carousel>
      </div>
    </div>
  );
};
