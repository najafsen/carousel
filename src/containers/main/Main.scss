$main-width: 1280px;
$header-height: 200px;
$carousel-height: 500px;

.main-component {
  display: flex;
  flex: 0 0 $main-width;
  flex-direction: column;

  .header-container {
    flex: 0 0 $header-height;
  }

  .carousel-container {
    display: flex;
    flex: 0 0 $carousel-height;
    max-height: $carousel-height;
    max-width: $main-width;

    .slide-content {
      display: flex;
      flex: 1;
      background-color: black;
      color: white;
    }
  }
}
