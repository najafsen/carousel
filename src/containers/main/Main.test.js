import React from "react";
import { render } from "@testing-library/react";
import { Main } from "./Main";

describe("Main", () => {
  test("matches snapshot", () => {
    const { asFragment } = render(<Main />);
    expect(asFragment()).toMatchSnapshot();
  });
});
