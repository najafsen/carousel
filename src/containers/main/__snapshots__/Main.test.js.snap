// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-component"
  >
    <div
      class="header-container"
    />
    <div
      class="carousel-container"
    >
      <div
        class="carousel-component"
        data-testid="carousel"
      >
        <div
          class="carousel-slide-component"
          style="margin-left: -000%; transition: margin-left 500ms ease-in-out;"
        >
          <div
            class="slide-content"
          >
            <div
              class="slide1"
            >
              <img
                alt="Comma Green"
                src="comma-green.svg"
              />
              <h1>
                This piece of software is 
                <i>
                  the shit
                </i>
                ,
                <br />
                I’ve never tried anything like it.
              </h1>
              <img
                alt="face"
                class="bottom-image"
                src="face.svg"
              />
              <h4>
                Mr. Face
              </h4>
              <p>
                Karma Guru @ Face Co™
              </p>
            </div>
          </div>
        </div>
        <div
          class="carousel-slide-component"
          style="transition: margin-left 500ms ease-in-out;"
        >
          <div
            class="slide-content"
          >
            <div
              class="slide2"
            >
              <img
                alt="Comma Red"
                src="comma-red.svg"
              />
              <h1>
                An experience that makes us feel like we’re covered in maple syrup –
                <i>
                   It’s a good thing
                </i>
              </h1>
              <img
                alt="Cake"
                class="bottom-image"
                src="cake.svg"
              />
              <h4>
                The whole team @
              </h4>
              <h4>
                The Pancake Collective™
              </h4>
            </div>
          </div>
        </div>
        <div
          class="carousel-slide-component"
          style="transition: margin-left 500ms ease-in-out;"
        >
          <div
            class="slide-content"
          >
            <div
              class="slide3"
            >
              <h1>
                These slides can contain anything, with any layout! Like previous slides.
              </h1>
            </div>
          </div>
        </div>
        <div
          class="carousel-slide-component"
          style="transition: margin-left 500ms ease-in-out;"
        >
          <div
            class="slide-content"
          >
            <div
              class="slide4"
            >
              <h1>
                Or, for example, 
                <span>
                  Tivoli!
                </span>
              </h1>
            </div>
          </div>
        </div>
        <div
          class="carousel-slide-component"
          style="transition: margin-left 500ms ease-in-out;"
        >
          <div
            class="slide-content"
          >
            <div
              class="slide5"
            >
              <h1>
                Or the view from my window in the morning! :)
              </h1>
            </div>
          </div>
        </div>
        <div
          class="navigation-buttons"
          data-testid="navigation-buttons"
        >
          <div
            class="navigation-button active"
          />
          <div
            class="navigation-button"
          />
          <div
            class="navigation-button"
          />
          <div
            class="navigation-button"
          />
          <div
            class="navigation-button"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
